doc/rst/technotes/libraries.rst

```diff
@@ -111,23 +205,145 @@ example looks like this:
    myCProg: myCProg.c lib/libfoo.a
      $(CHPL_COMPILER) $(CHPL_CFLAGS) -o myCProg myCProg.c $(CHPL_LDFLAGS)
 
+Makefile-less Compilation
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Alternatively, you can get the right combination is by using the ``compileline
+Alternatively, you can get the right combination by using the ``compileline
 --compile`` and ``compileline --libraries`` tools we provide.  The compilation
-command would then look like this (replacing myprog.c with the name of your C
-program):
+command would then look like this (replacing myCProg.c with the name of your C
+program that will use the library):
 
 .. code-block:: sh
 
-   `$CHPL_HOME/util/config/compileline --compile` myprog.c -L. -lfoo `$CHPL_HOME/util/config/compileline --libraries`
+   `$CHPL_HOME/util/config/compileline --compile` myCProg.c -Llib/ -lfoo `$CHPL_HOME/util/config/compileline --libraries`
 
 Note that ``compileline --compile-c++`` is also available. It requests a
 compilation command able to compile a C++ program.
 
+Using Your Library in Python
+============================
+
+Pre-requisites
+--------------
+
+To make use of your library in Python with minimal work, the Chapel compiler
+requires that you have ``python3`` installed in your ``$PATH``, and that your
+``python3`` installation contains ``numpy`` and ``Cython``.
+
+We don't anticipate supporting ``python2`` at this time, sorry.
+
+Compiling Your Chapel Library
+-----------------------------
+
+In addition to compiling with ``--library``, it is recommended to compile with
+``--library-python``.  This will create a Python-compatible module in addition
+to the normally generated library and header file.
+
+Python Module Name
+------------------
+
+By default, the name of the generated Python module will match the base name
```

Comment:
We use `basename` (one word) elsewhere, so maybe stay consistent with `basename`